core/contract.go

```diff
@@ -160,7 +160,7 @@ func (c *Contract) UpdateStorage(diff []StorageDiff, cb OnValueChanged) error {
 	}
 
 	// update contract storage root in the database
-	rootKeyDBKey := db.ContractRootKey.Key(c.Address.Marshal())
+	rootKeyDBKey := db.ContractStorage.Key(c.Address.Marshal())
```

Comment:
This is prone to corrupting the DB.

Contract storage tries use `db.ContractStorage.Key(c.Address.Marshal())` as it's prefix

https://github.com/NethermindEth/juno/blob/5670d77fe656667141be980916f908db17c832a8/core/contract.go#L226

and the rest of the key is essentially arbitrary;

https://github.com/NethermindEth/juno/blob/5670d77fe656667141be980916f908db17c832a8/core/trie/transaction_storage.go#L42-L52

if `key.WriteTo` generates the same byte array as `c.Address.Marshal()`, which is not likely but possible, could cause us to misinterpret the root key as a storage value.

Reply:
I got a little mixed up while thinking about this, so correct me if I'm wrong:

The result of `key.WriteTo` is appended to `t.prefix`, which includes `c.Address.Marshal()`. Thus, the `db.ContractStorage+c.Address` key is unused (since `key.WriteTo` cannot be empty) and we are free to use it for the root key.